### PR TITLE
🎨 Issue#1/ft website header

### PR DESCRIPTION
**Pull request by jbdura**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes # (issue) -Issue #3 💄  Added the header component

## Type of change

Please delete options that are not relevant.

- [ x] New feature (non-breaking change which adds functionality)

## Screenshot (optional)

Screenshot from 2023-08-09 19-09-05
Screenshot from 2023-08-09 18-52-27
Screenshot from 2023-08-09 18-51-52

## Checklist

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] My changes generate no new warnings
- [ x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LemmyMwaura** [comment]
Update your PR description. I'll review after that is done

**Comment by jbdura** [comment]
Edited the pr description

File: libs/elements/layout/header/src/lib/header/header.component.scss [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    .hamburger-icon {
+      display: none;
+      /* Hide by default for larger screens */
+      flex-direction: column;
+      cursor: pointer;
+
+      .bar {
+        width: 30px;
+        height: 3px;
+        background-color: #333;
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.scss:25]: Colours always at root level

File: libs/elements/layout/header/src/lib/header/header.component.scss [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        background-color: white;
+        box-shadow: 0px 4px 8px rgba(185, 183, 183, 0.1);
+        z-index: 100;
+        width: 150px;
+
+        li {
+          padding: 10px 20px;
+
+          &:hover {
+            background-color: #f5f5f5;
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.scss:70]: Direct colour

File: libs/elements/layout/header/src/lib/header/header.component.scss [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+    .dropdown {
+      position: relative;
+
+      .dropdown-list {
+        display: none;
+        position: absolute;
+        top: 100%;
+        left: 0;
+        background-color: white;
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.scss:61]: Direct colour

File: libs/elements/layout/header/src/lib/header/header.component.scss [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+      &:hover .dropdown-list {
+        display: inline-flex;
+      }
+    }
+
+      }
+    }
+
+    .contact-button {
+      background-color: #010e3a;
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.scss:84]: Direct colour

File: libs/elements/layout/header/src/lib/header/header.component.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'elewa-website-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+})
+export class HeaderComponent {
+  /**
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.ts:9]: This JsDoc should be on top of the class, else it won';t work

File: libs/elements/layout/header/src/lib/header/header.component.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+export class HeaderComponent {
+  /** 
+  * This is a header component class. 
+  * With this class, you can link the header section
+  */
+
+  // url for the logo
+  logoUrl = "https://res.cloudinary.com/dyl3rncv3/image/upload/v1675690297/elewa-group-website/Icons/SVG/_Logo/Logo_White_arvoyx.svg"
+
+  // To check if navigation menu is open or not
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.ts:17]: JsDoc

File: libs/elements/layout/header/src/lib/header/header.component.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  isMenuOpen = false;
+  toggleMenu() {
+    this.isMenuOpen = !this.isMenuOpen;
+  }
+
+  constructor () {
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.ts:23]: Don't put constructor if empty.

Constructor should be on top of normal methods

File: libs/elements/layout/header/src/lib/header/header.component.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  styleUrls: ['./header.component.scss'],
+})
+export class HeaderComponent {
+  /** 
+  * This is a header component class. 
+  * With this class, you can link the header section
+  */
+
+  // url for the logo
+  logoUrl = "https://res.cloudinary.com/dyl3rncv3/image/upload/v1675690297/elewa-group-website/Icons/SVG/_Logo/Logo_White_arvoyx.svg"
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.ts:15]: Perhaps for now, define as a constant at file level

File: libs/elements/layout/header/src/lib/header/header.component.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { Component } from '@angular/core';
+
```
Comment (jrosseel) [inline libs/elements/layout/header/src/lib/header/header.component.ts:2]: const LOGO_URL = .....

File: libs/elements/layout/header/src/lib/header/header.component.ts (hunk 8 above)
Comment (jrosseel) [reply libs/elements/layout/header/src/lib/header/header.component.ts:15]: logoUrl = LOGO_URL

**Review by jrosseel — CHANGES_REQUESTED** [review]

**Comment by jbdura** [comment]
Made the required changes.

**Comment by LemmyMwaura** [comment]
We are closing this as it's been resolved in #26. Suppose you need feedback reply below.

**Pull request closed without merging**